pkg/v6/resource/chartconfig/desired.go

```diff
@@ -19,6 +19,8 @@ const (
 	chartConfigVersionBundleVersion = "0.3.0"
 )
 
+type chartConfigGenerator func(ctx context.Context, clusterConfig cluster.Config, projectName string) (*v1alpha1.ChartConfig, error)
```

Comment:
We have these definitions usually in `spec.go`.

Reply:
Also with unexported types? Or is it ok to export it, even though its just used locally?

Reply:
I am personally cool with exported by default because of godoc support but Mr. Pawel has another opinion because of the package interface.

Reply:
Intuitively i would have said to not export such a local type we only use for the pattern here. But no strong opinions. I'll make anything to make you happy <3

Reply:
Not blocking here. Just think we should have these specifications in our usual `spec.go`. You got approvals. Go ahead as you like.